mono/mini/aot-runtime.c

```diff
@@ -6050,6 +6052,14 @@ mono_aot_get_unbox_trampoline (MonoMethod *method, gpointer addr)
 		return mono_aot_get_unbox_arbitrary_trampoline (addr);
 	}
 
+	if (!amodule->unbox_tramp_per_method) {
+		gpointer arr = g_new0 (gpointer, amodule->info.nmethods);
+		mono_memory_barrier ();
+		amodule->unbox_tramp_per_method = (gpointer*)arr;
```

Comment:
We should CAS here and free the old array if somebody else beat us to publishing the cache array. I'm not quite sure whether it is a problem if we register twice a tramp info (and twice a jit info) for the same code. We might have to lock and check if code was already published, and only register the tramp info once. I guess we would hav had this problem already for other trampolines if that's the case. Can't seem to find right now where we are handling concurrent compilations of the same methods with jit, for reference.